source/common/config/grpc_mux_impl.cc

```diff
@@ -12,10 +14,13 @@ GrpcMuxImpl::GrpcMuxImpl(const LocalInfo::LocalInfo& local_info, Grpc::AsyncClie
                          Event::Dispatcher& dispatcher,
                          const Protobuf::MethodDescriptor& service_method,
                          Runtime::RandomGenerator& random, Stats::Scope& scope,
-                         const RateLimitSettings& rate_limit_settings)
+                         const RateLimitSettings& rate_limit_settings,
+                         Server::ConfigTracker& config_tracker)
     : GrpcStream<envoy::api::v2::DiscoveryRequest, envoy::api::v2::DiscoveryResponse, std::string>(
           std::move(async_client), service_method, random, dispatcher, scope, rate_limit_settings),
-      local_info_(local_info) {
+      local_info_(local_info), config_tracker_entry_(config_tracker.add(
+                                   "control_plane", [this] { return dumpControlPlaneConfig(); })),
```

Comment:
I do not think that is possible with out ConfigTracker machinery. You need to register this to tracker then only it calls when `/config_dump' is accessed. May be @mattklein123  knows if there is any other way.

Reply:
The config tracker could be registered elsewhere, and there could be separate machinery to keep track of this state. Will this even work though for multiple xDS connections though? Won't this get registered multiple times if the user is not using ADS? I think this probably needs some rethinking?

Reply:
I thought about multiple xDS implementations. My thought was that it works similar to how `control_plane.connected_state` stat is working. Regarding duplicate, I think based on what I saw in config_tracker only one will be registered, others will not be added - Is that not correct?

Reply:
I'm not sure, you would need to look at the code. As I recall it will not register multiple times. Also, I don't think last one wins here makes sense. We would want to print the information for all connections?

Reply:
I am fine with that. I can change it to print with all connections and register them multiple times prefixing/suffixing the xDS it is registering. Is that fine?

Reply:
I would have to think it through, don't have a great answer for you. :) As long as print for all connections I think it should be fine. I would try to have an integration test for this. I suspect it wouldn't be that hard given some of the other xDS and config dump tests we have.

Reply:
@mattklein123  I thought through a littile bit and I think most easy and generic way is to build `ControlPlaneConfigDump` per each service like below. So I ADS is used, only first one will come. If user uses individual services, it gets repeated for each service - Do you think it is fine? If yes, I will make changes accordingly 

```
  {
   "@type": "type.googleapis.com/envoy.admin.v2alpha.ControlPlaneConfigDump",
     "service": "envoy.service.discovery.v2.AggregatedDiscoveryService",
     "control_plane": {
	    "identifier" : "control_plane_1"
	 },
     "last_updated": "1970-01-01T00:00:00Z"
  },
  {
   "@type": "type.googleapis.com/envoy.admin.v2alpha.ControlPlaneConfigDump",
     "service": "envoy.api.v2.EndpointDiscoveryService",
     "control_plane": {
	    "identifier" : "control_plane_1"
	 },
     "last_updated": "1970-01-01T00:00:00Z"
  }
```

Reply:
But I think it's possible there are multiple connections for each service? Let's say you have 2 services, each using an independent RDS connection? I think that is possible currently? (I can't actually remember if we multiplex all RDS over a single xDS connection, we might.)

Reply:
Sorry . Did follow fully - Are you saying that different Listeners can provide different RDS config sources?

Reply:
I think so but I can't remember. I would have a look. cc @AndresGuedez

Reply:
Yes, independent listeners can have independent RDS config source and connections.